### ISSUE TITLE
Work progress

### ISSUE THREAD

**Issue by LinerSRT** [LinerSRT/android_device_x5pro#7]
## Status:
- [x] "__xlog_buf_printf" fix
- [x] add libmrdump.so
- [x] turn off GateKeeper HAL
- [x] add all not located symbols:
    - [x] UCNV_FROM_U_CALLBACK_STOP_55
    - [x] _ZN7android13GraphicBufferC1Ejjij
    - [x] _ZN7android11BufferQueue...
    - [x] surfaceflinger mising symbols

- [ ] Fatal signal 11 (SIGSEGV), code 1, fault addr 0xc in tid 503 (audioserver) ## **LOG** https://gist.github.com/SeriniTY320/9531a9c32ee25770abe044b4a209a012

 -----------------------------
- Error LOG: https://gist.github.com/SeriniTY320/664aaa3d78aa86c4d3eb693121d93335#file-oreo_mt6735_log_error
- Sepolicy Errors: https://gist.github.com/SeriniTY320/664aaa3d78aa86c4d3eb693121d93335#file-oreo_mt6735_sepolicy_error
-----------------------------

**Comment by SeriniTY320** [comment]
+ CANNOT LINK EXECUTABLE "/system/bin/mobile_log_d": library "libmrdump.so" not found
Fixed. Commit will be soon

**Comment by SeriniTY320** [comment]
+ CANNOT LINK EXECUTABLE "/system/bin/mtk_agpsd": cannot locate symbol "UCNV_FROM_U_CALLBACK_STOP_55"
Temporary fixed (gps off)

**Comment by SeriniTY320** [comment]
+  CANNOT LINK EXECUTABLE "/system/bin/MtkCodecService": cannot locate symbol "__xlog_buf_printf" 
Liblog patched. Fixed

**Comment by SeriniTY320** [comment]
+  android.hardware.gatekeeper@1.0-service: Unable to open GateKeeper HAL
Turn off GateKeeper HAL

**Comment by LinerSRT** [comment]
Logcat before	**FIX: Add new symbols; Add new LD_PRELOAD; Add audio** and after **Liblog patch and using Libxlog shim**  commit:

- E HAL     : load: module=/system/lib/hw/hwcomposer.mt6735m.so 
E HAL     : dlopen failed: cannot locate symbol "_ZN7android11BufferQueue17createBufferQueueEPNS_2spINS_22IGraphicBufferProducerEEEPNS1_INS_22IGraphicBufferConsumerEEERKNS1_INS_19IGraphicBufferAllocEEE" referenced by "/system/lib/libgui_ext.so"...

- E HAL     : load: module=/system/lib/hw/camera.mt6735m.so
E HAL     : dlopen failed: cannot locate symbol "_ZN7android13GraphicBufferC1Ejjij" referenced by "/system/lib/libcam_utils.so"...

- F libc    : CANNOT LINK EXECUTABLE "/system/bin/mtk_agpsd": cannot locate symbol "UCNV_FROM_U_CALLBACK_STOP_55" referenced by "/system/bin/mtk_agpsd"...

**Comment by LinerSRT** [comment]
Logcat before: **Liblog patch and using Libxlog shim** : 

- E libEGL  : load_driver(/system/lib/egl/libGLES_mali.so): dlopen failed: cannot locate symbol "__xlog_buf_printf" referenced by "/system/lib/libdpframework.so"...
- F libEGL  : couldn't find an OpenGL ES implementation

- E HAL     : load: module=/system/lib/hw/audio.primary.mt6735m.so
- E HAL     : dlopen failed: library "libmrdump.so" not found

- E FMLIB_CORE: FMR_get_cfgs failed, dlopen failed: cannot locate symbol "__xlog_buf_printf" referenced by "/system/lib/libfmcust.so"...

- E RILD    : dlopen failed: dlopen failed: library "libmrdump.so" not found

- F libc    : CANNOT LINK EXECUTABLE "/system/bin/mtk_agpsd": cannot locate symbol "UCNV_FROM_U_CALLBACK_STOP_55" referenced by "/system/bin/mtk_agpsd

- F libc    : CANNOT LINK EXECUTABLE "/system/bin/guiext-server": cannot locate symbol "__xlog_buf_printf" referenced by "/system/bin/guiext-server"

- android.hardware.gatekeeper@1.0-service: Unable to open GateKeeper HAL

**Comment by LinerSRT** [comment]
logcat after shyming: 
09-28 06:00:49.797503   503   503 F libc    : Fatal signal 11 (SIGSEGV), code 1, fault addr 0xc in tid 503 (audioserver)

**Comment by mohancm** [comment]
did you fixed those symbols?

**Comment by LinerSRT** [comment]
Yes. Now problem is crashing audioserver (libaudiopolicymanager crash)

**Comment by mohancm** [comment]
You got boot animation?

**Comment by LinerSRT** [comment]
Nope. Stuck on logo

**Comment by WisniaPL** [comment]
Hi, how do you fix rebooting to fastboot ?

**Comment by mohancm** [comment]
@WisniaPL through binder commits for kernel

**Comment by manjotsidhu** [comment]
:face_with_thermometer:  Now did it booted ?
